### DISCUSSION TITLE
Hydrogen storefront (Shopify's framework)

### DISCUSSION THREAD

**Discussion by mariusa** [medusajs/medusa#2932] [Feature Requests]
Since Shopify is developing an open source storefront
https://shopify.dev/custom-storefronts/hydrogen
https://github.com/Shopify/hydrogen
https://github.com/Shopify/hydrogen-ui

how about having a starter storefront based on it too? 

Ideally, if there would be Shopify API compatibility built into medusajs, the Hydrogen storefront would just work by pointing it to a medusa store instead of a Shopify store.

Shopify's storefront is new but already better than nextjs starter (has search, navigation at top...). It will continue to benefit from 
1. updates & functionality from the best in this market (for comparison, I've been watching Vue Storefront for 4 years. I'd love to use it, but it's still a clunky mess with extremely slow progress).
2. themes & styling variations

and it will ensure API compatibility with Shopify (storefront). I think this is critical if you'd like to win customers over from Shopify. Merchants can keep their Hydrogen storefront as is and just switch the backend. Otherwise nobody will disrupt their functional shop with a completely different solution. 

Like it or not, Shopify is by far the leading industry player. You can't battle it on all fronts at once. If you can adopt parts of it (storefront, backend admin API compatiblity), it will make possible migration for some merchants.

**Comment by isinghdev** [comment]
Medusajs already has a Nextjs starter template for a storefront. What benefit would this provide over it?
Starter template link: https://github.com/medusajs/nextjs-starter-medusa

**Reply by mariusa** [reply]
I already tested the nextjs storefront.

Besides Shopify's storefront being better (has search, navigation at top...), it will continue to benefit from 
1. updates & functionality from the best in this market (for comparison, I've been watching Vue Storefront for 4 years. I'd love to use it, but it's still a clunky mess with extremely slow progress).
2. themes & styling variations

and it will ensure API compatibility with Shopify (storefront). I think this is critical if you'd like to win customers over from Shopify. Merchants can keep their Hydrogen storefront as is and just switch the backend. Otherwise nobody will disrupt their functional shop with a completely different solution. 

Like it or not, Shopify is by far the leading industry player. You can't battle it on all fronts at once. If you can adopt parts of it (storefront, backend admin API compatiblity), it will make possible migration for some merchants.
